storage/transfermanager/downloader.go

```diff
@@ -82,29 +152,46 @@ func (d *Downloader) downloadWorker() {
 		// Only request partSize data to begin with. If no error and we haven't finished
 		// reading the object, enqueue the remaining pieces of work and mark in the
 		// out var the amount of shards to wait for.
-		d.output <- input.downloadShard(d.client, d.config.perOperationTimeout)
+		out := input.downloadShard(d.client, d.config.perOperationTimeout)
+
+		// Keep track of any error that occurred.
+		if out.Err != nil {
+			d.error(out.Err)
```

Comment:
Can we wrap this error to include the bucket/object name? I don't see that added in downloadShard either.

Reply:
Sounds good; yeah if it's in the top-level output then you can see which individual result to inspect as well.